src/main/scala/pricemigrationengine/dynamodb/DynamoDBZIO.scala

```diff
@@ -0,0 +1,62 @@
+package pricemigrationengine.dynamodb
+
+import java.util
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB
+import com.amazonaws.services.dynamodbv2.model.{AttributeValue, QueryRequest, QueryResult}
+import pricemigrationengine.services.Logging
+import zio.console.Console
+import zio.stream.ZStream
+import zio.{IO, UIO, ZIO, ZLayer}
+
+import scala.jdk.CollectionConverters._
+
+case class DynamoDBZIOError(message: String)
+
+object DynamoDBZIO {
+  trait Service {
+    def query[A](
+      query: QueryRequest,
+      deserializer: java.util.Map[String, AttributeValue] => IO[DynamoDBZIOError, A]
+    ): ZStream[Any, DynamoDBZIOError, A]
+  }
+
+  val impl:  ZLayer[DynamoDBClient with Logging, Nothing, DynamoDBZIO] =
```

Comment:
Perhaps keep the implementation separate from the interface so that it's clearer to distinguish.  I know that in the ZIO docs they seem to put the main implementation with the spec but might be clearer apart.